weblate/api/serializers.py

```diff
@@ -113,6 +114,87 @@ def validate_code(self, value):
         return value
 
 
+class UserSerializer(serializers.ModelSerializer):
+    groups = serializers.HyperlinkedIdentityField(
+        view_name="api:group-detail", lookup_field="id", many=True, read_only=True,
+    )
+
+    class Meta:
+        model = User
+        fields = (
+            "email",
+            "full_name",
+            "username",
+            "groups",
+            "is_superuser",
+            "is_active",
+            "date_joined",
+            "url",
+        )
+        extra_kwargs = {
+            "url": {"view_name": "api:user-detail", "lookup_field": "username"}
+        }
+
+
+class PermissionSerializer(serializers.RelatedField):
+    class Meta:
+        model = Permission
+
+    def to_representation(self, value):
+        return value.name
+
+
+class RoleSerializer(serializers.ModelSerializer):
+    permissions = PermissionSerializer(read_only=True, many=True)
+
+    class Meta:
+        model = Role
+        fields = (
+            "name",
+            "permissions",
+        )
+
+    def validate_name(self, value):
+        check_query = Role.objects.filter(name=value)
+        if not check_query.exists():
+            raise serializers.ValidationError("Role with this role name was not found.")
+        return value
+
+
+class GroupSerializer(serializers.ModelSerializer):
+    roles = RoleSerializer(many=True)
```

Comment:
Going again through the patch, I'm not sure this is good approach as we want to add roles to the API in the future (to be able to manage permissions). Having something different in place right now will break the API once permission management will be added. So I'd either remove roles altogether for now, or have read only roles viewset + serializer which will be extended in future without breaking what would be introduced right now.

Reply:
True. I was conflicted about this too. maybe can keep read-only like `source_language` was from `Projects` till recently.

Reply:
Yes, having that read only for now is probably simplest approach.

Reply:
Done!